### ISSUE TITLE
Print all output even if last line is not followed by a newline with Bash 3.2

### ISSUE THREAD

**Issue by steff1193** [bats-core/bats-core#145]
Reopen of https://github.com/bats-core/bats-core/issues/141

```
$ uname
Darwin
$ bash -version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)
Copyright (C) 2007 Free Software Foundation, Inc.
$ bats -v
Bats 1.1.0
$ cat <<EOF > test.bats
> #!/usr/bin/env bats
> 
> @test "demo output with no newline is not shown" {
>     echo "line 1"
>     echo -n "line 2"
>     [ 1 -eq 2 ]
> }
> EOF
$ chmod 755 test.bats
$ ./test.bats
 ✗ demo output with no newline is not shown
   (in test file test.bats, line 7)
     `[ 1 -eq 2 ]' failed
   line 1

1 test, 1 failure
$ rm test.bats
$
```

**Comment by mbland** [comment]
Hmm, I thought GitHub would've let you reopen #141 instead of opening a new one. Quick search shows that isn't the case when a repo collaborator closed them.

Sorry about that, and about the assumption that v1.0.1 fixed the issue. That said, thanks for providing this level of detail in this new issue.

I am able to reproduce what you're seeing; apparently the "output printed even when no final newline" test case isn't triggering this behavior on 3.2.57(1)-release. I can't jump into it right now, but I will fix it, replacing the existing test case with the one you've provided.

**Comment by dd-ssc** [comment]
Any news on this ?

**Comment by martin-schulze-vireso** [comment]
I think this has been fixed with #99 so I am closing this now. Cannot reproduce the problematic output from the OP anymore.

**Comment by martin-schulze-vireso** [comment]
Okay, I did not notice that this is only a problem with Bash 3.2, reopening.
